src/openepd/model/specs/generated/asphalt.py

```diff
@@ -0,0 +1,86 @@
+#
+#  Copyright 2024 by C Change Labs Inc. www.c-change-labs.com
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#  This software was developed with support from the Skanska USA,
+#  Charles Pankow Foundation, Microsoft Sustainability Fund, Interface, MKA Foundation, and others.
+#  Find out more at www.BuildingTransparency.org
+#
+from openepd.compat.pydantic import pyd
+from openepd.model.specs.base import BaseOpenEpdHierarchicalSpec
+from openepd.model.specs.generated.enums import AsphaltGradation, AsphaltMixType
+from openepd.model.validation.numbers import RatioFloat
+from openepd.model.validation.quantity import LengthMmStr, TemperatureCStr, validate_unit_factory
+
+
+class AsphaltV1(BaseOpenEpdHierarchicalSpec):
+    """Asphalt performance specification."""
+
+    _EXT_VERSION = "1.0"
+
+    # Own fields:
+    aggregate_size_max: LengthMmStr | None = pyd.Field(default=None, description="Max aggregate size", example="20 mm")
+    rap: RatioFloat | None = pyd.Field(
+        default=None,
+        description="Percent of mixture that has been replaced by recycled asphalt pavement (RAP).",
+        example=0.5,
+        ge=0,
```

Comment:
Are you intentionally duplicate `ge=0`? It is already implied by `RatioFloat`.

Reply:
Yes. For now, while we are on pydantic 1 in cqd_etl, the annotated types are not fully supported, and their validators do not work. When we switch to 2, we could clean up the validators in field initializers, and leave only with the semantic types.